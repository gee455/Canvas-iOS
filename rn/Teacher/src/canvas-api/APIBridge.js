//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import {
  NativeModules,
  NativeEventEmitter,
} from 'react-native'

import api from './'
import i18n from 'format-message'

const { APIBridge } = NativeModules

export async function handleAPIBridgeRequest (body: any) {
  const { requestID, args, name } = body
  const apiCall = api[name]
  if (!apiCall) {
    throw new Error(`${name} cannot be found in the API bridge. Check the name of the api you are trying to call.`)
  }

  try {
    let result = await apiCall(...args)
    return APIBridge.requestCompleted(requestID, result.data, null)
  } catch (e) {
    return APIBridge.requestCompleted(requestID, null, i18n('Request could not be completed.'))
  }
}

export default function setup () {
  const apiBridgeEmitter = new NativeEventEmitter(APIBridge)
  apiBridgeEmitter.addListener('APICall', handleAPIBridgeRequest)
}
