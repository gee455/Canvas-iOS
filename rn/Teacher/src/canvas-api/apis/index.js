//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import * as accountApi from './account'
import * as coursesApi from './courses'
import * as usersApi from './users'
import * as enrollmentsApi from './enrollments'
import * as submissionsApi from './submissions'
import * as assignmentsApi from './assignments'
import * as quizzesApi from './quizzes'
import * as assignmentGroupsApi from './assignmentGroups'
import * as discussionsApi from './discussions'
import * as conversationApi from './conversations'
import * as groupsApi from './groups'
import * as loginApi from './login'
import * as externalTools from './external-tools'
import * as mediaComments from './media-comments'
import * as fileUploads from './file-uploads'
import * as files from './files'
import * as accounts from './accounts'
import * as userCustomData from './user-custom-data'

export default {
  ...accountApi,
  ...coursesApi,
  ...usersApi,
  ...enrollmentsApi,
  ...groupsApi,
  ...submissionsApi,
  ...assignmentsApi,
  ...quizzesApi,
  ...assignmentGroupsApi,
  ...discussionsApi,
  ...conversationApi,
  ...loginApi,
  ...externalTools,
  ...mediaComments,
  ...fileUploads,
  ...files,
  ...accounts,
  ...userCustomData,
}
