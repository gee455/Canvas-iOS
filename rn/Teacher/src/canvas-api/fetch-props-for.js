//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
import hoistStatics from 'hoist-non-react-statics'
import React, { type ComponentType } from 'react'
import { API, httpCache } from './model-api'

export type FetchProps = {
  api: API,
  isLoading: boolean,
  isSaving: boolean,
  loadError: ?Error,
  saveError: ?Error,
  refresh: () => void,
}

export function fetchPropsFor<OrigProps, QueryProps> (
  View: ComponentType<{ ...OrigProps, ...QueryProps, ...FetchProps }>,
  query: (OrigProps, API) => QueryProps
) {
  type State = { ...QueryProps, ...FetchProps }
  class Wrapper extends React.PureComponent<OrigProps, State> {
    loads: Set<ApiPromise<any>> = new Set()
    saves: Set<ApiPromise<any>> = new Set()
    softRefreshTimer: TimeoutID | void
    isReady: boolean = false

    update (state: $Shape<State>) {
      if (this.isReady) {
        this.setState(state)
      } else {
        this.state = { ...this.state, ...state }
      }
    }

    refresh = () => {
      this.update(query(this.props, this.refreshApi))
    }

    handleLoadStart = (request: ApiPromise<any>) => {
      if (this.loads.size === 0) {
        this.update({ isLoading: true, loadError: null })
      }
      this.loads.add(request)
    }

    handleLoadComplete = (request: ApiPromise<any>) => {
      this.loads.delete(request)
      this.update(query(this.props, this.cachedApi))
      if (this.loads.size === 0) {
        this.update({ isLoading: false })
      }
    }

    handleLoadError = (loadError: Error) => {
      this.update({ loadError })
    }

    handleSaveStart = (request: ApiPromise<any>) => {
      if (this.saves.size === 0) {
        this.update({ isSaving: true, saveError: null })
      }
      this.saves.add(request)
    }

    handleSaveComplete = (request: ApiPromise<any>) => {
      this.saves.delete(request)
      this.update(query(this.props, this.cachedApi))
      if (this.saves.size === 0) {
        this.update({ isSaving: false })
      }
    }

    handleSaveError = (saveError: Error) => {
      this.update({ saveError })
    }

    unsubscribe = httpCache.subscribe((request: ?ApiPromise<any>) => {
      if (request && (this.loads.has(request) || this.saves.has(request))) return
      clearTimeout(this.softRefreshTimer)
      this.softRefreshTimer = setTimeout(() => {
        this.softRefreshTimer = undefined
        this.update(query(this.props, this.api)) // refetch if invalidated
      }, 1000)
    })

    api = new API({
      policy: 'cache-and-network',
      onStart: this.handleLoadStart,
      onComplete: this.handleLoadComplete,
      onError: this.handleLoadError,
    })

    refreshApi = new API({
      policy: 'network-only',
      onStart: this.handleLoadStart,
      onComplete: this.handleLoadComplete,
      onError: this.handleLoadError,
    })

    saveApi = new API({
      policy: 'network-only',
      onStart: this.handleSaveStart,
      onComplete: this.handleSaveComplete,
      onError: this.handleSaveError,
    })

    cachedApi = new API({
      policy: 'cache-only',
    })

    state = {
      ...query(this.props, this.api),
      api: this.saveApi,
      isLoading: this.loads.size > 0,
      isSaving: this.saves.size > 0,
      loadError: null,
      saveError: null,
      refresh: this.refresh,
    }

    componentWillMount () {
      this.isReady = true
    }

    componentWillUnmount () {
      // prevent calling setState on unmounted component
      this.isReady = false
      this.unsubscribe()
      this.api.cleanup()
      this.refreshApi.cleanup()
      this.saveApi.cleanup()
      this.loads.clear()
      this.saves.clear()
      clearTimeout(this.softRefreshTimer)
      this.softRefreshTimer = undefined
    }

    render () {
      return <View {...this.props} {...this.state} />
    }
  }
  Wrapper.displayName = View.displayName || View.name
  return hoistStatics(Wrapper, View)
}
