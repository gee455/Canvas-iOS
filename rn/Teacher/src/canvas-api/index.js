//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import 'babel-polyfill'

export { default } from './apis/index'
export * from './apis/assignmentGroups'
export * from './apis/assignments'
export * from './apis/conversations'
export * from './apis/courses'
export * from './apis/discussions'
export * from './apis/enrollments'
export * from './apis/external-tools'
export * from './apis/groups'
export * from './apis/login'
export * from './apis/quizzes'
export * from './apis/submissions'
export * from './apis/users'
export * from './apis/media-comments'
export * from './apis/file-uploads'
export * from './apis/files'
export * from './apis/accounts'

export * from './session'

export { default as httpClient, isAbort, httpCache } from './httpClient'
