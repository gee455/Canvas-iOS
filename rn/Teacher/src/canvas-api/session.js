//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

let currentSession: ?Session = null

export function setSession (session: ?Session): void {
  currentSession = session
}

export function getSession (): Session {
  if (!currentSession) {
    console.warn('The user session is being requested, yet none exists. Are you sure that is what you want?')
    // The main usecase for actually returning this blank session is when components render on logout
    // On logout, we clear out redux which triggers components to render, which in turn those components might need a session
    // But, blank session is fine because the user has logged out
    return {
      baseURL: '',
      authToken: '',
      user: {
        id: '',
        primary_email: '',
        avatar_url: '',
        name: '',
      },
    }
  }
  return currentSession
}

// There are a few places where we do need to get the session and know whether it exists, but that's rare
// Almost everywhere should use getSession() instead
export function getSessionUnsafe (): ?Session {
  return currentSession
}

// returns true is the sessions are similar, otherwise returns false
export function compareSessions (s1: Session, s2: Session): boolean {
  return (
    s1.baseURL === s2.baseURL &&
    s1.user.id === s2.user.id &&
    s1.actAsUserID === s2.actAsUserID
  )
}
