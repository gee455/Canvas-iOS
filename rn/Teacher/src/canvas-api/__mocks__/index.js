//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import api from '../apis/index'
import { getSession, setSession, getSessionUnsafe } from '../session'
import { default as httpClient, isAbort, httpCache } from './httpClient'

const moduleExports = {
  default: api,
  httpClient,
  httpCache,
  isAbort,
  getSession,
  setSession,
  getSessionUnsafe,
}

Object.keys(api).forEach((functionName) => {
  moduleExports[functionName] = jest.fn()
})

module.exports = (moduleExports: CanvasApi)
