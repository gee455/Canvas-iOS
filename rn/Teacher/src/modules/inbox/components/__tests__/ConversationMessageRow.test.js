//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/* eslint-disable flowtype/require-valid-file-annotation */
import 'react-native'
import React from 'react'
import ConversationMessage from '../ConversationMessageRow'
import { getSession, setSession } from '../../../../canvas-api'
import explore from '../../../../../test/helpers/explore'

const template = {
  ...require('../../../../__templates__/conversations'),
  ...require('../../../../__templates__/users'),
  ...require('../../../../__templates__/helm'),
  ...require('../../../../__templates__/session'),
}

jest
  .mock('TouchableHighlight', () => 'TouchableHighlight')
  .mock('TouchableOpacity', () => 'TouchableOpacity')
  .mock('../../../../common/components/Avatar', () => 'Avatar')

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const convo = template.conversation({
    id: '1',
  })
  const message = template.conversationMessage()

  const tree = renderer.create(
    <ConversationMessage conversation={convo} message={message} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with author being the logged in user', () => {
  const session = getSession()
  const convo = template.conversation({
    id: '1',
    participants: [
      { id: session.user.id, name: 'hey there i am bob' },
      { id: '99999999', name: 'hey there i am jane' },
    ],
    audience: ['99999999'],
  })
  const message = template.conversationMessage({
    author_id: session.user.id,
  })

  const tree = renderer.create(
    <ConversationMessage conversation={convo} message={message} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with author lots of participants', () => {
  const session = getSession()

  const convo = template.conversation({
    id: '1',
    participants: [
      { id: session.user.id, name: 'hey there i am bob' },
      { id: '1', name: 'hey there i am jane' },
      { id: '2', name: 'hey there i am joe' },
      { id: '3', name: 'hey there i am jim' },
    ],
    audience: ['1', '2', '3'],
  })
  const message = template.conversationMessage({
    author_id: session.user.id,
  })

  const tree = renderer.create(
    <ConversationMessage conversation={convo} message={message} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('navigates to compose when reply to first message button pressed', () => {
  const session = template.session({
    user: {
      ...template.session().user,
      id: '1',
    },
  })
  setSession(session)
  const conversation = template.conversation({
    id: '1',
    participants: [
      { id: '1', name: 'hey there i am bob' },
      { id: '2', name: 'hey there i am joe' },
      { id: '3', name: 'hey there i am jim' },
    ],
    audience: ['2', '3'],
    context_name: 'Course 1',
    context_code: 'course_1',
    subject: 'Subject 1',
  })
  const navigator = template.navigator({ show: jest.fn() })
  const message = template.conversationMessage({
    author_id: session.user.id,
  })
  const props = {
    conversation,
    navigator,
    message,
    firstMessage: true,
  }
  const tree = renderer.create(
    <ConversationMessage {...props} />
  ).toJSON()
  const replyButton: any = explore(tree).selectByID('inbox.conversation-message-row.reply-button')
  replyButton.props.onPress()
  expect(navigator.show).toHaveBeenCalledWith(
    '/conversations/1/add_message',
    { modal: true },
    {
      recipients: [
        { id: '2', name: 'hey there i am joe' },
        { id: '3', name: 'hey there i am jim' },
      ],
      contextName: 'Course 1',
      contextCode: 'course_1',
      subject: 'Subject 1',
      canSelectCourse: false,
      canEditSubject: false,
      navBarTitle: 'Reply',
    },
  )
})

it('navigates to context card when the avatar is pressed', () => {
  const convo = template.conversation({
    id: '1',
  })
  const message = template.conversationMessage()
  const navigator = template.navigator()

  let view = renderer.create(
    <ConversationMessage navigator={navigator} conversation={convo} message={message} />
  ).toJSON()
  let avatar = explore(view).selectByType('Avatar')
  avatar.props.onPress()

  expect(navigator.show).toHaveBeenCalledWith(
    `/courses/1/users/1234`,
    { modal: true, modalPresentationStyle: 'currentContext' },
  )
})

it('navigates to a link when pressed', () => {
  const convo = template.conversation({})
  const message = template.conversationMessage()
  const navigator = template.navigator()

  let instance = renderer.create(
    <ConversationMessage navigator={navigator} conversation={convo} message={message} />
  ).getInstance()

  const link = 'http://www.google.com'
  instance.handleLink(link)

  expect(navigator.show).toHaveBeenCalledWith(link, { deepLink: true })
})
