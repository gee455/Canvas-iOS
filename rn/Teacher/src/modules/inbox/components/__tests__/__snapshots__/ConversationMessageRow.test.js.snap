// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  testID="inbox.conversation-message-123456789"
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderTopColor": "#C7CDD1",
        "borderTopWidth": 0.5,
        "flex": 1,
        "padding": 16,
        "paddingTop": 12,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#C7CDD1",
          "borderBottomWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 12,
        }
      }
    >
      <View
        accessibilityLabel="artorias of the abyss to me May 31 at 11:59 PM"
        accessible={true}
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "height": 32,
              "marginRight": 8,
              "width": 32,
            }
          }
        >
          <Avatar
            avatarURL={undefined}
            height={32}
            onPress={[Function]}
            userName="artorias of the abyss"
          />
        </View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                undefined,
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 14,
                    "fontWeight": "600",
                  },
                  Object {
                    "fontFamily": ".SFUIDisplay-semibold",
                  },
                ]
              }
            >
              artorias of the abyss 
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#8B969E",
                    "fontSize": 14,
                  },
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                ]
              }
            >
              to me
            </Text>
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#8B969E",
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
              ]
            }
          >
            May 31 at 11:59 PM
          </Text>
        </View>
      </View>
      <TouchableOpacity
        accessibilityLabel="Message options"
        accessibilityTraits="button"
        accessible={true}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "height": 32,
            "justifyContent": "center",
            "width": 32,
          }
        }
        testID="conversation-message.kabob-123456789"
        underlayColor="#ffffff00"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/kabob.png",
            }
          }
          style={
            Object {
              "height": 18,
              "tintColor": "#8B969E",
              "transform": Array [
                Object {
                  "rotate": "180deg",
                },
              ],
              "width": 18,
            }
          }
        />
      </TouchableOpacity>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "paddingTop": 16,
        }
      }
      testID={undefined}
    >
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          this is neat
        </Text>
      </View>
    </View>
    <TouchableOpacity
      onPress={[Function]}
      testID="inbox.conversation-message-123456789.attachment-1"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginTop": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/attachment.png",
            }
          }
          style={
            Object {
              "tintColor": "#008EE2",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#008EE2",
                "fontFamily": ".SFUIDisplay-bold",
                "fontSize": 14,
                "marginLeft": 6,
              },
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          something.pdf
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`renders correctly with author being the logged in user 1`] = `
<View
  testID="inbox.conversation-message-123456789"
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderTopColor": "#C7CDD1",
        "borderTopWidth": 0.5,
        "flex": 1,
        "padding": 16,
        "paddingTop": 12,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#C7CDD1",
          "borderBottomWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 12,
        }
      }
    >
      <View
        accessibilityLabel="me to hey there i am jane May 31 at 11:59 PM"
        accessible={true}
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "height": 32,
              "marginRight": 8,
              "width": 32,
            }
          }
        >
          <Avatar
            avatarURL={undefined}
            height={32}
            onPress={[Function]}
            userName="hey there i am bob"
          />
        </View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                undefined,
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 14,
                    "fontWeight": "600",
                  },
                  Object {
                    "fontFamily": ".SFUIDisplay-semibold",
                  },
                ]
              }
            >
              me 
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#8B969E",
                    "fontSize": 14,
                  },
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                ]
              }
            >
              to hey there i am jane
            </Text>
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#8B969E",
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
              ]
            }
          >
            May 31 at 11:59 PM
          </Text>
        </View>
      </View>
      <TouchableOpacity
        accessibilityLabel="Message options"
        accessibilityTraits="button"
        accessible={true}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "height": 32,
            "justifyContent": "center",
            "width": 32,
          }
        }
        testID="conversation-message.kabob-123456789"
        underlayColor="#ffffff00"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/kabob.png",
            }
          }
          style={
            Object {
              "height": 18,
              "tintColor": "#8B969E",
              "transform": Array [
                Object {
                  "rotate": "180deg",
                },
              ],
              "width": 18,
            }
          }
        />
      </TouchableOpacity>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "paddingTop": 16,
        }
      }
      testID={undefined}
    >
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          this is neat
        </Text>
      </View>
    </View>
    <TouchableOpacity
      onPress={[Function]}
      testID="inbox.conversation-message-123456789.attachment-1"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginTop": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/attachment.png",
            }
          }
          style={
            Object {
              "tintColor": "#008EE2",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#008EE2",
                "fontFamily": ".SFUIDisplay-bold",
                "fontSize": 14,
                "marginLeft": 6,
              },
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          something.pdf
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`renders correctly with author lots of participants 1`] = `
<View
  testID="inbox.conversation-message-123456789"
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderTopColor": "#C7CDD1",
        "borderTopWidth": 0.5,
        "flex": 1,
        "padding": 16,
        "paddingTop": 12,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#C7CDD1",
          "borderBottomWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 12,
        }
      }
    >
      <View
        accessibilityLabel="me to 3 others May 31 at 11:59 PM"
        accessible={true}
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "height": 32,
              "marginRight": 8,
              "width": 32,
            }
          }
        >
          <Avatar
            avatarURL={undefined}
            height={32}
            onPress={[Function]}
            userName="hey there i am bob"
          />
        </View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                undefined,
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 14,
                    "fontWeight": "600",
                  },
                  Object {
                    "fontFamily": ".SFUIDisplay-semibold",
                  },
                ]
              }
            >
              me 
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#8B969E",
                    "fontSize": 14,
                  },
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                ]
              }
            >
              to 3 others
            </Text>
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#8B969E",
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
              ]
            }
          >
            May 31 at 11:59 PM
          </Text>
        </View>
      </View>
      <TouchableOpacity
        accessibilityLabel="Message options"
        accessibilityTraits="button"
        accessible={true}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "height": 32,
            "justifyContent": "center",
            "width": 32,
          }
        }
        testID="conversation-message.kabob-123456789"
        underlayColor="#ffffff00"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/kabob.png",
            }
          }
          style={
            Object {
              "height": 18,
              "tintColor": "#8B969E",
              "transform": Array [
                Object {
                  "rotate": "180deg",
                },
              ],
              "width": 18,
            }
          }
        />
      </TouchableOpacity>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "paddingTop": 16,
        }
      }
      testID={undefined}
    >
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          this is neat
        </Text>
      </View>
    </View>
    <TouchableOpacity
      onPress={[Function]}
      testID="inbox.conversation-message-123456789.attachment-1"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginTop": 8,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/attachment.png",
            }
          }
          style={
            Object {
              "tintColor": "#008EE2",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#008EE2",
                "fontFamily": ".SFUIDisplay-bold",
                "fontSize": 14,
                "marginLeft": 6,
              },
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          something.pdf
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;
